src/Minesweeper.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import reactDom from 'react-dom';
+import PropTypes from 'prop-types';
+import Board from './components/Board/Board';
+import BoardHead from './components/BoardHead/BoardHead';
+
+ class Minesweeper extends React.Component {
+
+    constructor(props){
+        super(props);
+     
+        this.state = {
+                rows: props.rows,
+                columns : props.columns,
+                flags: 0,
+                mines: props.mines,
+                time: 0,
+                status: 0, //at game starts: waiting, running, ended
+                openCells : 0
+
+
+
+        };
+        // this.openGameClick = this.openGameClick.bind(this);
+        this.increamentFlags = this.increamentFlags.bind(this);
+        this.increamentOpenCells = this.increamentOpenCells.bind(this);
+        this.updateStatus = this.updateStatus.bind(this);
+        this.startTimer = this.startTimer.bind(this);
+        this.tick = this.tick.bind(this);
+    
+    }
+
+
+
+    // a function that increse the time every second
+    tick = ()=>
```

Comment:
## Make sure to lint your code with a linter (can be done directly with vscode) for a better looking code